### ISSUE TITLE
KDE Plasma 5 to 6 upgrades do not work

### ISSUE THREAD

**Issue by Sebazzz** [LukeShortCloud/winesapOS#791]
**Describe the bug or feature**
When running a Winesapos upgrade it is possible to terminate the upgrade accidentally, leaving the system in a broken state.

**Issue Type**
- [x] Bug Report
- [ ] Feature Request
- [ ] Documentation
- [ ] Question
- [ ] Other [Please specify]

**To Reproduce**
1. Run winesapos upgrade
2. Press escape

**Expected behavior**
Window should not close, upgrade should continue.

Currently I have the system in a broken state. For instance, the init files were corrupted (corrupt zstd signature). I was able to fix this by chrooting into the partition and running `mkinitcpio -p linux-lts`

![image](https://github.com/LukeShortCloud/winesapOS/assets/1426097/10a211cb-eada-4f01-85c8-33dfd16c34a6)

I got the system at least booting now, but Plasma is crashing, supposedly due to the same cause.

**Comment by LukeShortCloud** [comment]
Hey @Sebazzz , thanks for reporting this!

The `kdialog` messages that appear with information about the upgrade actually have no affect on the upgrade which is running in the background. Did you reboot shortly after hitting the ESCAPE button? That might have caused some issues.

Another problem is that Arch Linux just upgraded KDE Plasma from 5 to 6 in the last 24 hours. I just learned about that an hour ago so I have not had time to test and add additional upgrade tasks for that. I do know that, from CI builds today, that building winesapOS from scratch is currently broken.

One thing we have been considering is to use the Arch Linux archive for winesapOS. This way, we pin winesapOS to a specific set of Arch Linux packages to ensure that the first-time setup and upgrade continue to work regardless of what happens in upstream Arch Linux. Refer to: https://github.com/LukeShortCloud/winesapOS/issues/773

What is the output of this command? I want to verify what version of KDE Plasma is installed now.

```
pacman -Q | grep plasma
```

Can you also upload your latest upgrade log? It should be stored at `/etc/winesapos/upgrade_<DATE_AND_TIME>.log`.

**Comment by Sebazzz** [comment]
Yup it is running Plasma 6:
![image](https://github.com/LukeShortCloud/winesapOS/assets/1426097/ecef494a-6b03-48ea-ac9d-3f5ec1315fbc)

(Also notice the first messages, something about "database winesapos already being registered")

**Comment by Sebazzz** [comment]
[logs.zip](https://github.com/LukeShortCloud/winesapOS/files/14527695/logs.zip)
Attached logs

**Comment by Sebazzz** [comment]
BTW: the login screen shows just fine, but afterwards we find the loader (as usual)
![image](https://github.com/LukeShortCloud/winesapOS/assets/1426097/2361eec2-ba43-4cd5-8d21-25c7b6f49b7d)

And after this these error messages:
![image](https://github.com/LukeShortCloud/winesapOS/assets/1426097/a5721fd7-7397-4f3e-a8f7-700151c26b5f)

**Comment by LukeShortCloud** [comment]
Thanks for providing the screenshots and logs, that is very helpful. I can review the logs later when I get off of work.

It appears that KDE Plasma 6.0.0 is broken in Arch Linux right now. As a workaround, you can install a different desktop environment such as Cinnamon. After installing it, select the wrench icon in the top-right to change the desktop environment to something else before logging in.

```
$ sudo pacman -S cinnamon maui-pix xed
```

**Comment by Sebazzz** [comment]
I'm going to try the approach, thanks. 

Until now I've tried to downgrade to the 2024/02/20 repo where plasma 5.59.0-2 was still a thing, and forced the downgrade with `pacman -Syudd --overwrite "*" plasma` but this resulted in no crash, but a empty desktop with a single cursor:

![image](https://github.com/LukeShortCloud/winesapOS/assets/1426097/333cfbc3-b927-44dc-b9e9-14b18bedeca5)

But installing cinnamon worked, so thats fine too! It seems to be quite a bit faster as well.

**Comment by LukeShortCloud** [comment]
I looked over the logs yesterday and noticed a few things:

1. `crudini` was removed. It was supposed to be replaced by a newer package but that failed. This is why you are seeing the warning that the database is already  registered because that command is used to remove old and duplicate entries.
    - Short-term solution: edit the `/etc/pacman.conf` file and ensure that there is only one `[winesapos]` section.
    - Long-term solution: https://github.com/LukeShortCloud/winesapOS/issues/781
2. `mkinitcpio -P` failed because not all packages were upgraded.
    - Short-term solution: run the upgrade again or chroot and manually fix it like you did.
    - Long-term solution: https://github.com/LukeShortCloud/winesapOS/issues/736
3. `plasma-meta` was upgraded to KDE Plasma 6.0.0. perfectly fine but it is broken in upstream Arch Linux.
    - Short-term solution: use a different desktop environment such as Cinnamon.
    - Long-term solution: wait for KDE Plasma 6.0.1 to hopefully address these issues.

----

Logs:

1. `crudind`

```
+ crudini --set /etc/pacman.conf core SigLevel Never
bash: regel 182: crudini: opdracht niet gevonden
```

2. `mkinitcpio -P`

```
==> Building image from preset: /etc/mkinitcpio.d/linux-lts.preset: 'default'
==> Using default configuration file: '/etc/mkinitcpio.conf'
  -> -k /boot/vmlinuz-linux-lts -g /boot/initramfs-linux-lts.img --microcode /boot/*-ucode.img
==> Starting build: '6.1.38-1-lts'
  -> Running build hook: [base]
  -> Running build hook: [udev]
  -> Running build hook: [block]
==> WARNING: Possibly missing firmware for module: 'aic94xx'
==> WARNING: Possibly missing firmware for module: 'wd719x'
==> WARNING: Possibly missing firmware for module: 'xhci_pci'
  -> Running build hook: [keyboard]
  -> Running build hook: [autodetect]
  -> Running build hook: [modconf]
  -> Running build hook: [filesystems]
  -> Running build hook: [fsck]
==> Generating module dependencies
==> Creating zstd-compressed initcpio image: '/boot/initramfs-linux-lts.img'
bsdtar: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by bsdtar)
bsdtar: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /usr/lib/libarchive.so.13)
bsdtar: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by bsdtar)
bsdtar: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /usr/lib/libarchive.so.13)
==> ERROR: Image generation FAILED: 'sort reported an error'
```

**Comment by LukeShortCloud** [comment]
The short summary is that for the upcoming winesapOS 4.1.0 release, I want to completely refactor our upgrade script to (1) work for any Arch Linux distribution and (2) have a lot more safeguards to prevent issues like these.

**Comment by LukeShortCloud** [comment]
More specifically, I think part of the issues with the initramfs not working is due to a new release of `mkinitcpio` that happened just a few days before your upgrade.

It states that all of these packages need to be upgraded at the same time. Otherwise, the generated initramfs will be broken:
- mkinitcpio 38-3
- systemd 255.4-2
- lvm2 2.03.23-3
- mdadm 4.3-2
- cryptsetup 2.7.0-3

https://archlinux.org/news/mkinitcpio-hook-migration-and-early-microcode/

**Comment by LukeShortCloud** [comment]
@Sebazzz Do you happen to have a NVIDIA graphics card? I have been reading reports that KDE Plasma 6 switches the default session from X11 to Wayland. This causes issues with NVIDiA if Linux was not booted with the `nvidia_drm.modeset=1` option. See if you can also switch between "Plasma (X11)" and "Plasma (Wayland)" using the wrench icon on the login page. Do either of those work?

https://www.reddit.com/r/archlinux/comments/1b87m2b/so_arch_linux_is_officially_the_first/

**Comment by Sebazzz** [comment]
No this laptop only has embedded Intel graphics; switching between X11 and Wayland didn't work; and during my time troubleshooting I attached the harddrive to my main computer and ran the OS in a Hyper-V VM.

Met vriendelijke groet,
Sebastiaan Dammann
________________________________
Van: Luke Short ***@***.***>
Verzonden: Wednesday, March 13, 2024 6:14:13 PM
Aan: LukeShortCloud/winesapOS ***@***.***>
CC: Sebastiaan Dammann ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [LukeShortCloud/winesapOS] Winesapos upgrade can be (accidentally) terminated by pressing Escape (Issue #791)


@Sebazzz<https://github.com/Sebazzz> Do you happen to have a NVIDIA graphics card? I have been reading reports that KDE Plasma 6 switches the default session from X11 to Wayland. This causes issues with NVIDiA if Linux was not booted with the nvidia_drm.modeset=1 option. See if you can also switch between "Plasma (X11)" and "Plasma (Wayland)" using the wrench icon on the login page. Do either of those work?

https://www.reddit.com/r/archlinux/comments/1b87m2b/so_arch_linux_is_officially_the_first/

**Comment by LukeShortCloud** [comment]
Even on a fresh install of KDE Plasma 6, Plasma 5 and Qt5 packages are installed. I looked it up and this is expected behavior. The KDE team is still porting packages over.

I tried removing unneeded packages but that did not help.

```
$ pacman -Qdtq | sudo pacman -R -n -s --noconfirm -
```

Looking at `journalctl`, the `plasma-plasmashell` user service keeps crashing. The crash dumps in the log file mean nothing to me. On the desktop, the crash report window does not show the actual error.

This is a really hard problem to solve.

**Comment by LukeShortCloud** [comment]
Switching from LightDM to SDDM does not help.

**Comment by LukeShortCloud** [comment]
I found some forums suggesting that (1) removing xf86-video-* X11 drivers and (2) removing old KDE Plasma configuration files may help.

https://bbs.archlinux.org/viewtopic.php?id=293795
https://discuss.kde.org/t/plasmashell-crashes-frequently-after-upgrading-to-plasma-6-0/11467

**Comment by LukeShortCloud** [comment]
@Sebazzz KDE Plasma 5 to 6 upgrades are now working based on testing I did today!

Arch Linux released KDE Plasma 6.0.3 today which seems to work (along with some recent winesapOS upgrade logic to deal with upgrading a few of the Plasma packages). Older versions (6.0.0, 6.0.1, and 6.0.2) I have tested to have upgrade issues.

Could you try doing an upgrade on your system using the commands below to see if KDE Plasma's Wayland session works afterwards? X11 may still have some issues. Plasma 5 had stable X11 and unstable Wayland support. Plasma 6 is, ironically, the opposite (unstable X11 and stable Wayland).

```
export WINESAPOS_UPGRADE_VERSION_CHECK=false
curl https://raw.githubusercontent.com/LukeShortCloud/winesapOS/main/scripts/winesapos-upgrade.sh | sudo -E bash
```

**Comment by LukeShortCloud** [comment]
I will close this out now. Please let me know if you run into any lingering issues.

**Comment by LukeShortCloud** [comment]
It turns out this was a wide-spread issue for KDE Plasma 5 to 6 upgrades. The KDE project fixed the migration in 6.0.3 but the original fix was to delete the local user cache.

```
$ rm -r -f ~/.cache/*
```

https://youtu.be/lvUQVFE5lIU?t=6853
